src/main/scala/is/hail/variant/Call.scala

```diff
@@ -7,77 +7,178 @@ import is.hail.utils._
 
 object Call extends Serializable {
 
-  def apply(call: Int): Call = {
-    require(call >= 0, s"Call must be null or >= 0. Found ${ call }.")
-    call
+  def apply(ar: Int): Call = {
+    require(ar >= 0, s"Call must be null or >= 0. Found ${ ar }.")
+    Call(ar, phased = false, ploidy = 2)
   }
 
-  def toString(call: Call): String = {
-      val p = Genotype.gtPair(call)
+  def apply(j: Int, k: Int, phased: Boolean): Call = {
+    val gt =
+      if (phased)
+        Genotype.gtIndexPhased(j, k)
+      else
+        Genotype.gtIndexWithSwap(j, k)
+
+    Call(gt, phased, ploidy = 2)
+  }
+
+  def apply(ar: Int, phased: Boolean, ploidy: Int): Call = {
+    require(ploidy == 2, s"invalid ploidy: $ploidy. Only support ploidy == 2")
+    require(ar >= 0, s"invalid allele representation: $ar. Must be positive.")
+
+    var c = 0
+    c |= phased.toInt
+
+    if (ploidy > 2)
+      c |= (3 << 1)
+    else
+      c |= (ploidy << 1)
+
+    require((ar >>> 29) == 0, s"invalid allele representation: $ar. Max value is 2^29 - 1")
+    c |= ar << 3
+
+    c
+  }
+
+  def isPhased(c: Call): Boolean = (c & 0x1) == 1
+
+  def isDiploid(c: Call): Boolean = ploidy(c) == 2
+
+  def isUnphasedDiploid(c: Call): Boolean = (c & 0x7) == 4
+
+  def isPhasedDiploid(c: Call): Boolean = (c & 0x7) == 5
+
+  def ploidy(c: Call): Int = (c >>> 1) & 0x3
+
+  def alleleRepr(c: Call): Int = c >>> 3
+
+  def gtPair(c: Call): GTPair = {
+    require(isDiploid(c), s"invalid ploidy: ${ ploidy(c) }. Only support ploidy == 2")
+
+    if (isPhased(c))
+      Genotype.gtPairPhased(alleleRepr(c))
+    else
+      Genotype.gtPair(alleleRepr(c))
+  }
+
+  def unphasedAlleleRepr(c: Call): Call = {
```

Comment:
I'm open to changing the name of this.

Reply:
Should the return value be a call?  I think it should match the constructor, maybe `unphasedDiploidTriangleNumber`?  Oof.

Also, I'm uncomfortable with this implicitly unphasing things.  I think it should assert if the call is not unphased diploid.  That way existing stuff works, but methods for which phasing hasn't been thought through will give a meaningful error message.